controllers/cluster_controller.go

```diff
@@ -374,6 +382,12 @@ func (r *ClusterReconciler) preClusterProviderReconcile(ctx context.Context, log
 		}
 	}
 
+	if cluster.IsManaged() {
+		if err := r.machineHealthCheck.Reconcile(ctx, log, cluster); err != nil {
```

Comment:
yes the machine health checks get deleted if the cluster is deleted. why do we want to wait till the capi cluster is created?

Reply:
I think we had this discussion where mhcs can get in the way of node rollouts, but not sure if that is an issue anymore.

Reply:
> why do we want to wait till the capi cluster is created?
Because if the capi cluster object doesn't exist, the MHC won't be deleted if the user deletes the eks-a cluster. What's the point of creating MHC for a cluster that we don't know if it will be created (it can fail validations for exmaple).

Reply:
I wanted to install them pre cluster reconcile so that while creating the machines for the workload cluster itself the machine health checks could have been leveraged. I agree with the delete use case tho, moved it to post reconcile.